src/niri.rs

```diff
@@ -910,6 +913,7 @@ impl Niri {
             ForeignToplevelManagerState::new::<State, _>(&display_handle, |client| {
                 !client.get_data::<ClientState>().unwrap().restricted
             });
+        let screencopy_state = ScreencopyManagerState::new::<State>(&display_handle);
```

Comment:
Needs a filter on `restricted` clients like above.

Reply:
I added this functionality, but i don't know how to test it. It seems fairly simple, though, and i've basically just copied what `foreign_toplevel.rs` does